### PR TITLE
[Feature](fe) Add admin set partition version statement

### PR DESCRIPTION
**Pull request by xy720**
## Proposed changes

Last week, we encountered a situation where we had to recover master's metadata from image without bdbje log.
After recovering the master, we found that the partition visible version of the master's metadata was inconsistent with the actual replica in Be.

This commit add a statement to modify partition visible version.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
some docs still need to be updated

**Comment by xy720** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.74 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162140296 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/persist/SetPartitionVersionOperationLog.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+public class SetPartitionVersionOperationLog implements Writable {
+
+    @SerializedName(value = "partitionId")
+    private long partitionId;
+    @SerializedName(value = "versionInfo")
+    private long versionInfo;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/persist/SetPartitionVersionOperationLog.java:35]:
```suggestion
    private long visibleVersion;
```

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -5379,6 +5381,62 @@ public void cleanTrash(AdminCleanTrashStmt stmt) {
...
+            LOG.warn("Failed to set partition visible version to {}. Partition {} not exists.",
+                    log.getVersionInfo(), log.getPartitionId());
+        }
+    }
+
+    public int setPartitionVersionInternal(long partitionId, long versionInfo, boolean isReplay) throws DdlException {
+        int result = -1;
+        List<Long> dbIds = getInternalCatalog().getDbIds();
+        // TODO should use inverted index
+        for (long dbId : dbIds) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:5406]: how about set dbid and table id by user?
Iterate all dbs and tables is very expansive.

**Review by morningman — COMMENTED** [review]
Please add regression test case.
Eg, if a partition's version is 3, 
1. set it to 2
2. set it to 3
3. check if table can query
4. load some data
5. check if table can query

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java (hunk 2 above)
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:5406]: Sure.

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.13 seconds
 stream load tsv:          542 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162305756 Bytes

**Comment by xy720** [comment]
@morningman

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.07 seconds
 stream load tsv:          542 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17161946626 Bytes

**Review by morningman — DISMISSED** [review]
LGTM

**Review by chenlinzhong — DISMISSED** [review]
LGTM

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run clickbench

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.97 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162209572 Bytes

**Review by chenlinzhong — APPROVED** [review]
LGTM

**Review by adonis0147 — APPROVED** [review]
LGTM

**Comment by xiaokang** [comment]
revert this PR in branch-2.0 since it modify fe metadata OperationType and cause rollback from 2.0.2 to 2.0.1/2.0.0

**Pull request merged**